pkg/querier/chunkstore/chunkstore.go

```diff
@@ -0,0 +1,15 @@
+package chunkstore
+
+import (
+	"context"
+
+	"github.com/cortexproject/cortex/pkg/chunk"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+)
+
+// ChunkStore is the read-interface to the Chunk Store.  Made an interface here
+// to reduce package coupling.
+type ChunkStore interface {
```

Comment:
Since go interfaces are implemented implicitly you usually want to declare them where you need them. 

I might be wrong but you could copy the interface where it was required or coupled. I think this would be better.

Reply:
I moved this code to prevent a circular pkg dependency after refactoring the `LazyQueryable` into its own package (in order for consumption by multiple other packages) . It currently exists here:
https://github.com/cortexproject/cortex/blob/f3d3ddba04d7ec30af68f3a11e7f7b25976a7c2f/pkg/querier/querier.go#L55-L59

I don't mind redefining it in multiple places if that's preferred -- let me know.

Reply:
No big opinion here, just giving you an alternative.